typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -339,100 +273,68 @@ export abstract class HyperlaneDeployer<
+          proxyAdmin.upgradeAndCall(
+            proxy.address,
+            implementation.address,
+            initData,
+          ),
         ),
     );
   }
 
   protected async deployProxy<C extends ethers.Contract>(
     chain: ChainName,
     implementation: C,
-    initArgs: Parameters<C['initialize']>,
     proxyAdmin: string,
-    deployOpts?: DeployOptions,
+    initializeArgs?: Parameters<C['initialize']>,
   ): Promise<C> {
-    const initData = implementation.interface.encodeFunctionData(
-      'initialize',
-      initArgs,
-    );
+    const initData = initializeArgs
+      ? implementation.interface.encodeFunctionData(
+          'initialize',
+          initializeArgs,
+        )
+      : '0x';
 
-    let proxy: TransparentUpgradeableProxy;
-    const provider = this.multiProvider.getProvider(chain);
-    const deployer = await this.multiProvider.getSignerAddress(chain);
     this.logger(`Deploying transparent upgradable proxy`);
-    if (
-      deployOpts &&
-      deployOpts.create2Salt &&
-      (await provider.getCode(CREATE2FACTORY_ADDRESS)) != '0x'
-    ) {
-      // To get consistent addresses with Create2, we need to use
-      // consistent constructor arguments.
-      // The three constructor arguments we need to configure are:
-      // 1. Proxy implementation: This will start as the Create2Factory
-      //    address, as it needs to be a contract address.
-      //    After we've taken over as the proxy admin, we will set it
-      //    to the proper address.
-      // 2. Proxy admin: This will start as the deployer
-      //    address. We will use this to initialize before rotating.
-      // 3. Initialization data: This will start as null, and we will
-      //    initialize our proxied contract manually.
-      const constructorArgs: Parameters<
-        TransparentUpgradeableProxy__factory['deploy']
-      > = [CREATE2FACTORY_ADDRESS, deployer, '0x'];
-
-      // deploy with static implementation, deployer admin, and init data for consistent addresses
-      proxy = await this.deployContractFromFactory(
-        chain,
-        new TransparentUpgradeableProxy__factory(),
-        'TransparentUpgradeableProxy',
-        constructorArgs,
-        deployOpts,
-      );
-      // upgrade and initialize with actual implementation and init data
-      await this.upgradeAndInitialize(
-        chain,
-        proxy,
-        implementation.address,
-        initData,
-      );
-      // rotate admin to the desired admin
-      await this.changeAdmin(chain, proxy, proxyAdmin);
-    } else {
-      const constructorArgs: Parameters<
-        TransparentUpgradeableProxy__factory['deploy']
-      > = [implementation.address, proxyAdmin, initData];
-      proxy = await this.deployContractFromFactory(
-        chain,
-        new TransparentUpgradeableProxy__factory(),
-        'TransparentUpgradeableProxy',
-        constructorArgs,
-        deployOpts,
-      );
-    }
```

Comment:
🫡

Reply:
doing this earlier would have saved me a lot of headaches

Reply:
Yeah....

Reply:
RIP in peace